static/src/javascripts/projects/common/modules/discussion/loader.js

```diff
@@ -0,0 +1,580 @@
+                this.user = resp.userProfile;
+
+                Id.getUserFromApi(user => {
+                    if (user && user.publicFields.username) {
+                        this.username = user.publicFields.username;
+                    }
+
+                    this.emit('user:loaded');
+                });
+            });
+        } else {
+            this.emit('user:loaded');
+        }
+    }
+
+    getDiscussionId(): ?string {
+        return this.elem.getAttribute('data-discussion-key');
+    }
+
+    getDiscussionClosed(): boolean {
+        return this.elem.getAttribute('data-discussion-closed') === 'true';
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    getCommentIdFromHash(): ?number {
+        const reg = /#comment-(\d+)/;
+        return reg.exec(window.location.hash)
+            ? parseInt(reg.exec(window.location.hash)[1], 10)
+            : null;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    setCommentHash(id: string): void {
+        window.location.replace(`#comment-${id}`);
+    }
+
+    commentPosted(...args: Array<mixed>): void {
+        this.removeState('truncated');
+        this.comments.addComment(...args);
+    }
+
+    initState(): void {
+        if (this.getDiscussionClosed()) {
+            this.setState('closed');
+        } else if (this.comments.isReadOnly()) {
+            this.setState('readonly');
+        } else if (Id.getUserFromCookie()) {
+            if (
+                this.user &&
+                this.user.privateFields &&
+                !this.user.privateFields.canPostComment
+            ) {
+                this.setState('banned');
+            } else {
+                this.setState('open');
+            }
+        } else {
+            this.setState('open');
+        }
+    }
+
+    isCommentable(): boolean {
+        // not readonly, not closed and user is signed in
+        const userCanPost =
+            this.user &&
+            this.user.privateFields &&
+            this.user.privateFields.canPostComment;
+        return (
+            userCanPost &&
+            !this.comments.isReadOnly() &&
+            !this.getDiscussionClosed()
+        );
+    }
+
+    ready(): void {
+        this.$topCommentsContainer = $('.js-discussion-top-comments');
+
+        this.initTopComments();
+        this.initMainComments();
+        this.initToolbar();
+        this.renderCommentCount();
+        this.initPagination();
+        this.initRecommend();
+
+        DiscussionAnalytics.init();
+
+        // More for analytics than anything
+        if (window.location.hash === '#comments') {
+            mediator.emit('discussion:seen:comments-anchor');
+        } else if (this.getCommentIdFromHash()) {
+            mediator.emit('discussion:seen:comment-permalink');
+        }
+
+        mediator.on(
+            'discussion:commentbox:post:success',
+            this.removeState.bind(this, 'empty')
```

Comment:
Could you remove the `bind(this)` by using `() => { this.removeState('empty') }` instead?